Sources/Boutique/Store.swift

```diff
@@ -54,9 +54,20 @@ public final class Store<Item: Codable & Equatable>: ObservableObject {
     /// or subscribe to it however they wish, but you desire making modifications to ``items``
     /// you must use ``insert(_:)-7z2oe``, ``remove(_:)-3nzlq``, or ``removeAll()-9zfmy``.
     @MainActor @Published public private(set) var items: [Item] = []
+    
+    /// This is the task that loads the `Item`s after the store is initialized.
+    ///
+    /// The user can ``try await store.setupTask.value`` to make sure the ``items``
+    /// have loaded when using the synchronous version of the ``init``.
+    public private(set) lazy var setupTask: Task<Void, Error> = Task { @MainActor in
```

Comment:
This is the one thing in the patch that I'm stumped on, I don't really understand the expected usage of `setupTask` external to this class.

I see you wrote
> The user can ``try await store.setupTask.value`` to make sure the ``items`` have loaded when using the synchronous version of the ``init``

But I'm not sure in practice how this would be used. Is the expectation that a user would query `setupTask` to see if all items have been loaded, or use it to trigger a full reload, or something else that I'm not considering? If it's either of those two then perhaps a different abstraction (a name change, a different function, etc) may be the right answer, or I may be misunderstanding the intended purpose.

Reply:
Yeah I'm right there with you, and honestly I'm not 100% sure how to make the API more clear. 

To step back a little, the current issue is that there's no way to know when the `items` have loaded (except for, afaik, `objectWillChange` observations, like in `Stored`). And storing/exposing this task opens up a couple of ways of handling that in different situations. 

My thought process is this:
1. If the `Store` is used directly and can be initialized/stored from an `async` context, then using the `async` `init` is the simplest option - that will create the store and return once `items` have loaded. 
2. If the `Store` is used directly, but cannot be initialized from an `async` context for some reason, and you still want to make sure `items` are retrieved after they've been loaded, you can do something like:
```
let store: Store<SomeItem> = .init(...) // non-async

func fetchItems() async -> [SomeItem] { 
  await store.setupTask.value
  return await store.items
}
```
3. And `@Stored` is an edge case of the 2nd point above, i.e. `Store` has to be created synchronously, except it handles changes internally using Combine. 

I'm open to other suggestions though in terms of naming/api. Maybe instead of exposing the task directly, it could be wrapped in `func itemsDidLoad() async` or something, so it'd be `await store.itemsDidLoad()` instead of `setupTask.value`?

Reply:
Thank you for the explanation of what you were thinking, that definitely helps me understand the problem more clearly.

Since the goal is to communicate whether or not the items have loaded, I do like the idea of exposing a computed property of along the lines of your suggested `func itemsDidLoad() async`. I don't usually like to expose implementation details such as the underlying tasks, which is why I'm leaning towards your proposed solution of only telling the user whether the items have loaded, as I can see the benefit of knowing that in an async world.

Could we consider a name like `storeHasInitialized`, `storeHasLoaded`, or `itemsHaveLoaded`  for the property, returning the has loaded state as a `Bool`? (I prefer `storeHasInitialized` because it implies that this operation should only occur once, but I'm also _very_ open to suggestions.)

Thanks again for taking the time to help me get where you're coming from!